File: 125-ValidPalindrome/125-ValidPalindrome.java
// Last updated: 7/19/2026, 10:24:22 PM
class Solution {
    public boolean isPalindrome(String s) {
        s=s.replaceAll("[^A-Za-z0-9]", "").toLowerCase();
        return check(0,s.length()-1,s);
    }
    
    public boolean check(int left,int right,String s){
        if(s=="")
            return true;
        if(left>=right)
            return true;
        if(s.charAt(left)!=s.charAt(right))
            return false;
        return check(left+1,right-1,s);
    }
}

File: 125-ValidPalindrome/125-ValidPalindrome.cpp
// Last updated: 7/19/2026, 10:24:16 PM
#include <bits/stdc++.h>

class Solution {
public:
    bool isPalindrome(string s) {
        string filtered = "";
        for (char c : s) {
            if (isalnum(c)) {
                filtered += tolower(c);
            }
        }
        
        string reversed = filtered;
        reverse(reversed.begin(), reversed.end());
        
        return filtered == reversed;
    }
};
